src/proposals/20250529/SparkEthereum_20250529.t.sol

```diff
@@ -280,6 +444,126 @@ contract SparkEthereum_20250529Test is SparkTestBase {
+        
+        assertEq(uniSUsds.balanceOf(Unichain.PSM3), 0);
+
+        // --- Step 1: Deposit 10m USDS and 10m sUSDS into the PSM ---
+
+        vm.startPrank(Unichain.ALM_RELAYER);
+        uniController.depositPSM(Unichain.USDS,  10_000_000e18);
+        uniController.depositPSM(Unichain.SUSDS, susdsDepositShares);  // $10m
+        vm.stopPrank();
+
+        IPSMLike psm = IPSMLike(Unichain.PSM3);
+
+        assertApproxEqAbs(psm.convertToAssetValue(psm.shares(Unichain.ALM_PROXY)), 20_000_000e18, 27);
+
+        assertEq(uniUsds.balanceOf(Unichain.ALM_PROXY), 90_000_000e18);
+        assertEq(uniUsds.balanceOf(Unichain.PSM3),      10_000_000e18);
+
+        assertApproxEqAbs(uniSUsds.balanceOf(Unichain.ALM_PROXY), susdsShares - susdsDepositShares, 1);  // $90m
+        assertApproxEqAbs(uniSUsds.balanceOf(Unichain.PSM3),      susdsDepositShares,               1);  // $10m
+
+        chainData[ChainIdUtils.Ethereum()].domain.selectFork();
+        vm.warp(mainnetTimestamp);
+
+        // --- Step 2: Mint and bridge 10m USDC to Unichain ---
+
+        uint256 usdcAmount = 10_000_000e6;
+        uint256 usdcSeed   = 1e6;
+
+        vm.startPrank(Ethereum.ALM_RELAYER);
+        mainnetController.mintUSDS(usdcAmount * 1e12);
+        mainnetController.swapUSDSToUSDC(usdcAmount);
+        mainnetController.transferUSDCToCCTP(usdcAmount, CCTPForwarder.DOMAIN_ID_CIRCLE_UNICHAIN);
+        vm.stopPrank();
+
+        chainData[ChainIdUtils.Unichain()].domain.selectFork();
+        vm.warp(mainnetTimestamp);
+
+        assertEq(uniUsdc.balanceOf(Unichain.ALM_PROXY), 0);
+        assertEq(uniUsdc.balanceOf(Unichain.PSM3),      usdcSeed);
+
+        _relayMessageOverBridges();
+
+        assertEq(uniUsdc.balanceOf(Unichain.ALM_PROXY), usdcAmount);
+        assertEq(uniUsdc.balanceOf(Unichain.PSM3),      usdcSeed);
+
+        // --- Step 3: Deposit 10m USDC into PSM3 ---
+
+        vm.startPrank(Unichain.ALM_RELAYER);
+        uniController.depositPSM(Unichain.USDC, usdcAmount);
+        vm.stopPrank();
+
+        assertEq(uniUsdc.balanceOf(Unichain.ALM_PROXY), 0);
+        assertEq(uniUsdc.balanceOf(Unichain.PSM3),      usdcSeed + usdcAmount);
+
+        // --- Step 4: Withdraw all assets from PSM3 ---
+
+        vm.startPrank(Unichain.ALM_RELAYER);
+        uniController.withdrawPSM(Unichain.USDS,  10_000_000e18);
+        uniController.withdrawPSM(Unichain.SUSDS, susdsDepositShares);  // $10m
+        uniController.withdrawPSM(Unichain.USDC,  usdcAmount);
+        vm.stopPrank();
+
+        assertEq(uniUsds.balanceOf(Unichain.PSM3),  0);
+        assertEq(uniSUsds.balanceOf(Unichain.PSM3), 0);
+
+        assertApproxEqAbs(uniUsdc.balanceOf(Unichain.PSM3), usdcSeed, 1);
+
+        assertEq(uniUsds.balanceOf(Unichain.ALM_PROXY),  100_000_000e18);
+        assertApproxEqAbs(uniSUsds.balanceOf(Unichain.ALM_PROXY), susdsShares, 1);
+
+        usdcAmount -= 1;  // Rounding
+        assertEq(uniUsdc.balanceOf(Unichain.ALM_PROXY), usdcAmount);
```

Comment:
This still needs change that we discussed @supercontracts

Reply:
added the changes as discussed.